src/v/archival/ntp_archiver_service.cc

```diff
@@ -261,7 +263,7 @@ ss::future<ntp_archiver::scheduled_upload> ntp_archiver::schedule_single_upload(
         .base_timestamp = upload.base_timestamp,
         .max_timestamp = upload.max_timestamp,
         .delta_offset = delta,
-        .ntp_revision = _partition->get_revision_id(),
+        .ntp_revision = model::revision_id(_rev()),
```

Comment:
quick question:

what's the difference between `ntp.get_initial_revision` and `partition.get_revision_id`. if there is a difference, this change to using `_rev` (ie ntp.initial_revision) seems to change the semantics of the code but the patch seems to suggest only a refactoring here.

Reply:
I think splitting it up is useful even in cases where people like me ask simple questions like this :) thanks for explaining!